### PR TITLE
:sparkles: Fakeclient: Add support for evictions

### PR DESCRIPTION
**Pull request by alvaroaleman**
This change adds support for evictions into the fakeclient. 
Prior to this, it would just error on any subresource creation.

/cc @vincepri @sbueringer

### CODE REVIEW

File: pkg/client/fake/client.go [hunk 1]
```diff
@@ -969,7 +972,15 @@ func (sw *fakeSubResourceClient) Get(ctx context.Context, obj, subResource clien
 }
 
 func (sw *fakeSubResourceClient) Create(ctx context.Context, obj client.Object, subResource client.Object, opts ...client.SubResourceCreateOption) error {
-	panic("fakeSubResourceWriter does not support create")
+	switch subResource.(type) {
```
Comment (sbueringer) [inline pkg/client/fake/client.go:975]: Would it make sense to store "subResource string" of the SubResource call (l.936) in the fakeSubResourceClient struct and then only implement this logic here if the subResource (string) is indeed "eviction"?

Comment (alvaroaleman) [reply pkg/client/fake/client.go]: Good point, updated, PTAL

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**